core/app/models/spree/order.rb

```diff
@@ -432,7 +432,10 @@ def finalize!
 
       touch :completed_at
 
-      deliver_order_confirmation_email unless confirmation_delivered?
+      Spree.event_bus.publish(
+        :order_confirmed,
+        Spree::Events::OrderConfirmedEvent.new(order: self)
```

Comment:
what about using configurable classes instead? `Spree::Config.order_confirmed_event_class.new(...)`

Reply:
I think we want to ensure we *have* a strong API, so anyone that wishes to count on these events through their app or extensions can expect events to look a certain way.
If someone wanted to write their own event class (that, for instance, supported more things), they could catch this one and re-broadcast a different one, or override the code that triggers the event.